### PR TITLE
Automatically install default content types

### PR DESCRIPTION
**Pull request by icc**
Will also make it easier to update all content types with the push of a button.

### CODE REVIEW

File: classes/framework.php [hunk 1]
```diff
@@ -110,6 +110,55 @@ public static function get_language() {
     }
 
     /**
+     * Make it easy to download and install H5P libraries.
+     *
+     * @param $onlyupdate Prevent install of new libraries
```
Comment (thomasmars) [inline classes/framework.php]: Missing type of param

File: classes/framework.php [hunk 2]
```diff
@@ -490,7 +492,18 @@ public function isInDevMode() {
     /**
      * Implements mayUpdateLibraries
```
Comment (thomasmars) [inline classes/framework.php]: Missing param

Comment (icc) [inline classes/framework.php]: Fixed in 453d3b9

File: classes/task/look_for_updates.php [hunk 3]
```diff
@@ -46,7 +48,41 @@ public function execute() {
...
+                                              $updatesurl;
+                $message->fullmessageformat = FORMAT_PLAIN;
+                $message->fullmessagehtml   = '<p>' . get_string('updatesavailablemsgpt1', 'mod_hvp') . '<br/>' .
+                                              get_string('updatesavailablemsgpt2', 'mod_hvp') . '</p>' .
+                                              '<p>' . get_string('updatesavailablemsgpt3', 'mod_hvp', '<b>' . date('Y-m-d', $update_available) . '</b>') . '<br/>' .
+                                              get_string('updatesavailablemsgpt4', 'mod_hvp', '<b>' . date('Y-m-d', $current_update) . '</b>') . '</p>' .
+                                              '<a href="' . $updatesurl .  '" target="_blank">' . $updatesurl . '</a>';
+                $message->smallmessage      = '';
+                $message->notification      = 1;
+                message_send($message);
```
Comment (thomasmars) [inline classes/task/look_for_updates.php]: I get:
`Error: lib/moodlelib.php email_to_user(): Could not instantiate mail function.
++ Error calling message processor email ++
* line 127 of \lib\classes\message\manager.php: call to debugging()
* line 88 of \lib\classes\message\manager.php: call to core\message\manager::send_message_to_processors()
* line 229 of \lib\messagelib.php: call to core\message\manager::send_message()
* line 81 of \mod\hvp\classes\task\look_for_updates.php: call to message_send()
* line 75 of \lib\cronlib.php: call to mod_hvp\task\look_for_updates->execute()`

It seems this requires sendmail on the server, are we sure we want to rely on users having that set up ?
I did not have it.

Comment (icc) [inline classes/task/look_for_updates.php]: I'm wondering if this is up to the one setting up Moodle. I think you can request to have the messages show up as a popups instead.

File: db/install.php [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+    $interface->mayUpdateLibraries(true);
+
+    // Fetch info about library updates
+    $core = \mod_hvp\framework::instance('core');
+    $core->fetchLibrariesMetadata();
+
+    // Download default libraries and try to install
+    $error = \mod_hvp\framework::downloadH5pLibraries();
+    if ($error !== null) {
+      self::messages('error', $error);
```
Comment (thomasmars) [inline db/install.php]: Cannot use self, since not inside a class. Should probably be \mod_hvp\framework::messages('error', $error);

Comment (icc) [inline db/install.php]: Indeed, copy-paste mistake. 67bd09c

**Pull request merged**